src/common/transformations/src/transformations/common_optimizations/fuse_rotary_positional_embeddings.cpp

```diff
@@ -427,6 +428,79 @@ ov::pass::RoPEFusionPreprocess::RoPEFusionPreprocess() {
     this->register_matcher(m, callback);
 }
 
+ov::pass::RoPEFusionVIT3D::RoPEFusionVIT3D() {
+    using namespace ov::op::util;
+    MATCHER_SCOPE(RoPEFusionVIT3D);
+
+    auto x = pattern::any_input(pattern::rank_equals(3));
+    auto x_or_cos1 = pattern::any_input(pattern::rank_equals(3));
+    auto x_or_cos2 = pattern::any_input(pattern::rank_equals(3));
+    auto t_sin = pattern::any_input(pattern::rank_equals(3));
+
+    auto varsplit = pattern::wrap_type<v1::VariadicSplit>({x, 2, {"half_ndims", "?"}});
+    varsplit->set_output_size(2);
+
+    auto int32_max = std::numeric_limits<std::int32_t>::max();
+
+    auto x2 = NewGenSlice(x, "half_ndims", int32_max, 1, 2);
+    auto x2neg = pattern::wrap_type<v1::Multiply>({x2 | varsplit->output(1), -1.0f}, {{"auto_broadcast", "numpy"}});
+    auto x1 = NewGenSlice(x, 0, "half_ndims", 1, 2);
+    auto x_rotate_half = pattern::wrap_type<v0::Concat>({x2neg, x1 | varsplit->output(0)}, {{"axis", -1}});
+
+    auto mul_cos = pattern::wrap_type<v1::Multiply>({x_or_cos1, x_or_cos2}, {{"auto_broadcast", "numpy"}});
+    auto mul_sin = pattern::wrap_type<v1::Multiply>({x_rotate_half, t_sin}, {{"auto_broadcast", "numpy"}});
+
+    auto result = pattern::wrap_type<v1::Add>({mul_cos, mul_sin}, {{"auto_broadcast", "numpy"}});
+
+    matcher_pass_callback callback = [OV_CAPTURE_CPY_AND_THIS](ov::pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+        auto root = m.get_match_root();
+
+        // check mul(x, cos) exists
+        Output<Node> v_cos;
+        if (pattern_map.at(x_or_cos1) == pattern_map.at(x)) {
```

Comment:
If with `|` of `mul_cos` suggested above ,this check here could be removed

Reply:
fixed